Add data augmentation to input_ds.py

aabpettit opened the pull request.

aabpettit committed:
Update input_ds.py

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -49,14 +52,14 @@ def parse_example(serialized_example):
 
 
         parsed_features = tf.io.parse_single_example(serialized_example, features=data_fields)
+        
```
nit: remove whitespace changes

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -93,6 +95,116 @@ def parse_example(serialized_example):
+    
+    
+    
+    globals  = ['global_mask', 'global_std', 'global_transit_mask', 'global_view', 'global_view_0.3', 'global_view_5.0', 'global_view_double_period', 'global_view_double_period_0.3', 'global_view_double_period_5.0']
```
I think we don't want to add noise to `mask`s or `std`

but we do want to reverse everything

evantey14 replied:
(also applicable to locals / secondaries)

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -93,6 +95,116 @@ def parse_example(serialized_example):
+    globals  = ['global_mask', 'global_std', 'global_transit_mask', 'global_view', 'global_view_0.3', 'global_view_5.0', 'global_view_double_period', 'global_view_double_period_0.3', 'global_view_double_period_5.0']
+    locals   =['local_mask','local_std','local_std_even','local_std_odd','local_view','local_view_0.3','local_view_5.0','local_view_even','local_view_half_period_std','local_view_odd']
+    secondaries = ['secondary_mask','secondary_std','secondary_view','secondary_view_0.3','secondary_view_5.0']
+    all  = ['global_mask', 'global_std', 'global_transit_mask', 'global_view', 'global_view_0.3', 'global_view_5.0', 'global_view_double_period', 'global_view_double_period_0.3', 'global_view_double_period_5.0','local_mask','local_std','local_std_even','local_std_odd','local_view','local_view_0.3','local_view_5.0','local_view_even','local_view_half_period_std','local_view_odd','secondary_mask','s ...
```
Might be able to pull these from `config`?

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -93,6 +95,116 @@ def parse_example(serialized_example):
+    def add_noise_global(image,y):
+        for a in globals: 
+            z = image[0][a]
+            noise = tf.random.normal(shape=z.get_shape(), mean=0.0, stddev=0.1, dtype=tf.float32)
```
I think we want to choose `stddev` in a more informed way:
1) calculated 1.48 * mad(z)
2) use `std` array?

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -93,6 +95,116 @@ def parse_example(serialized_example):
+    def add_noise_local(image,y):
+        for a in locals: 
+            z = image[0][a]
+            noise = tf.random.normal(shape=z.get_shape(), mean=0.0, stddev=0.1, dtype=tf.float32)
```
Hmm these noise arrays are independently generated, when in reality noise between views is correlated.

Maybe for now we just say this doesn't matter too much

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -93,6 +95,116 @@ def parse_example(serialized_example):
+        return image[0],image[1],image[2]
+    
+    
+    def reverse_global(image,x,y):
```
Probably when we `reverse`, we want to reverse everything, not just `global`, cause then the datapoint would be self inconsistent -- i think this means we don't need these individual reverse functions and could just have one that goes through all relevant series.

aabpettit committed:
Update input_ds.py

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -93,6 +95,75 @@ def parse_example(serialized_example):
+#    false_rand = false_rand.filter(lambda x,z: z==1)
+#    ds_new = ds_new.filter(lambda image,y: y ==True)
+    
+    features = list(configurations.final_alpha_0()['inputs']['features'].keys())
+    globals = [feature for feature in features if 'global' in feature]
+    locals = [feature for feature in features if 'local' in feature]
+    secondaries = [feature for feature in features if 'secondary' in feature]
+    all = globals+locals+secondaries
```
`all_without_masks = [feature for feature in all if 'mask' not in feature]`

evantey14 commented on astronet/astro_cnn_model/input_ds.py:
```diff
@@ -93,6 +95,75 @@ def parse_example(serialized_example):
+    def add_noise(image,y):
+        for a in array_noise:
+            z = image[0][a]
+            noise = 1.48*stats.median_absolute_deviation(z, scale=1)
```
I think we want to add `nan_policy='omit'` since we may have series that have a few nans in them

evantey14 replied:
I think also we still want to do `tf.random.normal` sample? otherwise we'll just return `z` shifted up by the same `noise` at every position?

aabpettit committed:
Update input_ds.py

Edits based on comments- should now be usable, can make changes based on specific dispositions or on all data

aabpettit committed:
Update input_ds.py
